### ISSUE TITLE
[KPN] Unexpected Error: MissingSchema

### ISSUE THREAD

**Issue by TonivdW** [dut-iptv/dut-iptv.github.io#50]
Fresh install of LibreELEC (9.2.6 Kodi 18.9) on a RPi 2B with only this plugin added (0.9.116)
Installation went fine, all dependencies seem to have installed correctly.
EPG and channel info loads ok

If I navigate to TV from the main menu and select a channel then the following message appears:
KPN ITV
Proxy is not correctly started, this ofthen happens after first installing or updating, please restart Kodi and try again

Upon (re)boot the following message is shown:
KPN ITV Error
check log for more information

In kodi.log this is logged:
```
ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <class 'requests.exceptions.MissingSchema'>
                                            Error Contents: Invalid URL '/USER/DEVICES': No schema supplied. Perhaps you meant http:///USER/DEVICES?
                                            Traceback (most recent call last):
                                              File "/storage/.kodi/addons/plugin.video.kpn/service.py", line 2, in <module>
                                                main()
                                              File "/storage/.kodi/addons/plugin.video.kpn/resources/lib/base/l5/service.py", line 62, in main
                                                startup()
                                              File "/storage/.kodi/addons/plugin.video.kpn/resources/lib/base/l5/service.py", line 50, in startup
                                                hourly(type=0)
                                              File "/storage/.kodi/addons/plugin.video.kpn/resources/lib/base/l5/service.py", line 19, in hourly
                                                api_get_session()
                                              File "/storage/.kodi/addons/plugin.video.kpn/resources/lib/api.py", line 42, in api_get_session
                                                download = api_download(url=devices_url, type='get', headers=None, data=None, json_data=False, return_json=True)
                                              File "/storage/.kodi/addons/plugin.video.kpn/resources/lib/base/l5/api.py", line 17, in api_download
                                                resp = getattr(session, type)(url, allow_redirects=allow_redirects, auth=auth)
                                              File "/storage/.kodi/addons/script.module.requests/lib/requests/sessions.py", line 546, in get
                                                return self.request('GET', url, **kwargs)
                                              File "/storage/.kodi/addons/plugin.video.kpn/resources/lib/base/l4/session.py", line 52, in request
                                                data = super(Session, self).request(method, url, **kwargs)
                                              File "/storage/.kodi/addons/script.module.requests/lib/requests/sessions.py", line 519, in request
                                                prep = self.prepare_request(req)
                                              File "/storage/.kodi/addons/script.module.requests/lib/requests/sessions.py", line 462, in prepare_request
                                                hooks=merge_hooks(request.hooks, self.hooks),
                                              File "/storage/.kodi/addons/script.module.requests/lib/requests/models.py", line 313, in prepare
                                                self.prepare_url(url, params)
                                              File "/storage/.kodi/addons/script.module.requests/lib/requests/models.py", line 387, in prepare_url
                                                raise MissingSchema(error)
                                            MissingSchema: Invalid URL '/USER/DEVICES': No schema supplied. Perhaps you meant http:///USER/DEVICES?
                                            -->End of Python script error report<--
```

I have triple checked my XS4ALL login credentials (subscriber number and pin)
Please advice how to resolve this.

**Comment by dut-iptv** [comment]
The settings file for KPN seems to have not been correctly downloaded. This should auto correct eventually butbyou can force a redownload of the settings file from the addon. Open the addon and go to settings -> force settings download. Try again after

**Comment by TonivdW** [comment]
Force settings download does not seem to resolve the issue.
In the log I find:
```
ERROR: GetDirectory - Error getting plugin://plugin.video.kpn/?_=_download_settings
ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.kpn/?_=_download_settings) failed
```

**Comment by J-o-h-n-M** [comment]
Same issue here

**Comment by J-o-h-n-M** [comment]
Here is the full log of starting and also trying to download the settings 👍 
`2020-12-23 11:49:37.698 T:19568  NOTICE: PVR Manager: Starting
2020-12-23 11:49:37.703 T:17744  NOTICE: PVR Manager: Started
2020-12-23 11:49:37.934 T:14776  NOTICE: EPG thread started
2020-12-23 11:49:39.222 T:13496   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <class 'requests.exceptions.MissingSchema'>
                                            Error Contents: Invalid URL '/USER/DEVICES': No schema supplied. Perhaps you meant http:///USER/DEVICES?
                                            Traceback (most recent call last):
                                              File "C:\Users\John\AppData\Roaming\Kodi\addons\plugin.video.kpn\service.py", line 2, in <module>
                                                main()
                                              File "C:\Users\John\AppData\Roaming\Kodi\addons\plugin.video.kpn\resources\lib\base\l5\service.py", line 62, in main
                                                startup()
                                              File "C:\Users\John\AppData\Roaming\Kodi\addons\plugin.video.kpn\resources\lib\base\l5\service.py", line 50, in startup
                                                hourly(type=0)
                                              File "C:\Users\John\AppData\Roaming\Kodi\addons\plugin.video.kpn\resources\lib\base\l5\service.py", line 19, in hourly
                                                api_get_session()
                                              File "C:\Users\John\AppData\Roaming\Kodi\addons\plugin.video.kpn\resources\lib\api.py", line 42, in api_get_session
                                                download = api_download(url=devices_url, type='get', headers=None, data=None, json_data=False, return_json=True)
                                              File "C:\Users\John\AppData\Roaming\Kodi\addons\plugin.video.kpn\resources\lib\base\l5\api.py", line 17, in api_download
                                                resp = getattr(session, type)(url, allow_redirects=allow_redirects, auth=auth)
                                              File "C:\Users\John\AppData\Roaming\Kodi\addons\script.module.requests\lib\requests\sessions.py", line 546, in get
                                                return self.request('GET', url, **kwargs)
                                              File "C:\Users\John\AppData\Roaming\Kodi\addons\plugin.video.kpn\resources\lib\base\l4\session.py", line 52, in request
                                                data = super(Session, self).request(method, url, **kwargs)
                                              File "C:\Users\John\AppData\Roaming\Kodi\addons\script.module.requests\lib\requests\sessions.py", line 519, in request
                                                prep = self.prepare_request(req)
                                              File "C:\Users\John\AppData\Roaming\Kodi\addons\script.module.requests\lib\requests\sessions.py", line 462, in prepare_request
                                                hooks=merge_hooks(request.hooks, self.hooks),
                                              File "C:\Users\John\AppData\Roaming\Kodi\addons\script.module.requests\lib\requests\models.py", line 313, in prepare
                                                self.prepare_url(url, params)
                                              File "C:\Users\John\AppData\Roaming\Kodi\addons\script.module.requests\lib\requests\models.py", line 387, in prepare_url
                                                raise MissingSchema(error)
                                            MissingSchema: Invalid URL '/USER/DEVICES': No schema supplied. Perhaps you meant http:///USER/DEVICES?
                                            -->End of Python script error report<--
2020-12-23 11:49:39.426 T:17720  NOTICE: AddOnLog: PVR IPTV Simple Client: PVRIptvData::LoadEPG EPG Loaded - 481 (ms)
2020-12-23 11:49:52.072 T:18568   ERROR: C:\Users\John\AppData\Roaming\Kodi\addons\script.module.fuzzywuzzy\lib\fuzzywuzzy\fuzz.py:11: UserWarning: Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning
                                              warnings.warn('Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning')
2020-12-23 11:49:54.703 T:14776  NOTICE: PVR Manager: Stopping
2020-12-23 11:49:55.468 T:14776  NOTICE: PVR Manager: Stopped
2020-12-23 11:49:56.710 T:17424   ERROR: XFILE::CDirectory::GetDirectory - Error getting plugin://plugin.video.kpn/
2020-12-23 11:49:56.721 T:19568  NOTICE: AddOnLog: PVR IPTV Simple Client: PVRIptvData::LoadPlayList PlayList Loaded - 2 (ms)
2020-12-23 11:49:56.721 T:19568  NOTICE: AddOnLog: PVR IPTV Simple Client: Loaded 176 channels.
2020-12-23 11:49:56.721 T:19568  NOTICE: PVR Manager: Starting
2020-12-23 11:49:56.723 T:20684   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.kpn/) failed
2020-12-23 11:49:56.727 T:16944  NOTICE: PVR Manager: Started
2020-12-23 11:49:56.727 T:19568  NOTICE: EPG thread started
2020-12-23 11:49:58.286 T:16308  NOTICE: AddOnLog: PVR IPTV Simple Client: PVRIptvData::LoadEPG EPG Loaded - 323 (ms)
2020-12-23 11:49:59.392 T:18568 WARNING: Attempt to use invalid handle 1
2020-12-23 11:50:19.007 T:20684 WARNING: Previous line repeats 6 times.
2020-12-23 11:50:19.007 T:20684   ERROR: Control 55 in window 10025 has been asked to focus, but it can't
2020-12-23 11:50:28.398 T:9464   ERROR: XFILE::CDirectory::GetDirectory - Error getting plugin://plugin.video.kpn/?_=_download_settings
2020-12-23 11:50:28.403 T:20684   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.kpn/?_=_download_settings) failed`

**Comment by J-o-h-n-M** [comment]
I did see that settings.json is available. 

I even went into api.py and changed all {api_url} to https://api.tv.kpn.com/101/1.2.0/A/nld/pctv/kpn  but that did not do the trick.

**Comment by dut-iptv** [comment]
Will be fixed in 117

**Comment by dut-iptv** [comment]
This bug was reproduced and fixed in 0.9.122 which is currently available in the beta repository.

0.9.122 will probably be available in the main repository in a couple of days.

**Comment by dut-iptv** [comment]
0.9.128 is available in the main repo and should fix this issue.

If you still encounter this or any other bug please post in #59.

Closing.
